sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessorClient.cs

```diff
@@ -0,0 +1,982 @@
+                    }
+                    EnsureNotRunningAndInvoke(() => _processMessage = value);
+                }
+            }
+
+            remove
+            {
+                Argument.AssertNotNull(value, nameof(ProcessMessageAsync));
+
+                if (_processMessage != value)
+                {
+                    throw new ArgumentException(Resources1.HandlerHasNotBeenAssigned);
+                }
+
+                EnsureNotRunningAndInvoke(() => _processMessage = default);
+            }
+        }
+
+        /// <summary>
+        ///   The event responsible for processing unhandled exceptions thrown while this processor is running.
+        ///   Implementation is mandatory.
+        /// </summary>
+        ///
+        [SuppressMessage("Usage", "AZC0002:Ensure all service methods take an optional CancellationToken parameter.", Justification = "Guidance does not apply; this is an event.")]
+        [SuppressMessage("Usage", "AZC0003:DO make service methods virtual.", Justification = "This member follows the standard .NET event pattern; override via the associated On<<EVENT>> method.")]
+        public event Func<ExceptionReceivedEventArgs, Task> ProcessErrorAsync
+        {
+            add
+            {
+                Argument.AssertNotNull(value, nameof(ProcessErrorAsync));
+
+                if (_processErrorAsync != default)
+                {
+                    throw new NotSupportedException(Resources1.HandlerHasAlreadyBeenAssigned);
+                }
+
+                EnsureNotRunningAndInvoke(() => _processErrorAsync = value);
+            }
+
+            remove
+            {
+                Argument.AssertNotNull(value, nameof(ProcessErrorAsync));
+
+                if (_processErrorAsync != value)
+                {
+                    throw new ArgumentException(Resources1.HandlerHasNotBeenAssigned);
+                }
+
+                EnsureNotRunningAndInvoke(() => _processErrorAsync = default);
+            }
+        }
+
+        /// <summary>
+        ///   Signals the <see cref="ServiceBusProcessorClient" /> to begin processing events.  Should this method be called while the processor
+        ///   is running, no action is taken.
+        /// </summary>
+        /// <param name="processingOptions"></param>
+        ///
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the start operation.  This won't affect the <see cref="ServiceBusProcessorClient" /> once it starts running.</param>
+        ///
+        /// exception cref="EventHubsException">Occurs when this <see cref="ServiceBusProcessorClient" /> instance is already closed./exception>
+        /// <exception cref="InvalidOperationException">Occurs when this method is invoked without <see cref="ProcessMessageAsync" /> or <see cref="ProcessErrorAsync" /> set.</exception>
+        ///
+        public virtual async Task StartProcessingAsync(
```

Comment:
You may want to take a look at the Event Hubs Processor.   Alexander just had a [PR](https://github.com/Azure/azure-sdk-for-net/pull/10065) that changed the structure of these a bit for some new analyzer that I wasn't aware of.

Reply:
It looks like that is related to EH having both sync and async versions for these methods, which raises the question of whether we need that for SB. I thought EH was avoiding sync methods, but maybe that was just for service methods?